File: AndroidStudy/login/src/main/java/com/example/login/Acticity/MainActivity.java
package com.example.login.Acticity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.login.R;

public class MainActivity extends AppCompatActivity {
    private Button btn_open;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViews();
        btn_open.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, loginActivity.class);
                startActivity(intent);
            }
        });
    }

    private void findViews() {
        btn_open = findViewById(R.id.btn_open);
    }
}

File: KotlinStudy/paging/src/main/java/com/example/paging/activity/MainActivity.kt
package com.example.paging.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.paging.adapter.DataRecycleViewAdapter
import com.example.paging.mvvm.MainActivityViewModel
import com.example.paging.databinding.ActivityMainBinding
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val manager = LinearLayoutManager(this)
        binding.rvData.layoutManager = manager
        val dataRecycleViewAdapter = DataRecycleViewAdapter()
        binding.rvData.adapter = dataRecycleViewAdapter
        binding.btnGet.setOnClickListener {
            lifecycleScope.launch {
                MainActivityViewModel().getData().collectLatest {
                    dataRecycleViewAdapter.submitData(it)
                }
            }
        }
    }
}